com.github.Eloston.ungoogled-chromium.yaml

```diff
@@ -0,0 +1,160 @@
+        url: https://github.com/apple/cups/releases/download/v2.3.3/cups-2.3.3-source.tar.gz
+        sha256: 261fd948bce8647b6d5cb2a1784f0c24cc52b5c4e827b71d726020bcc502f3ee
+
+  - name: gtk-cups-backend
+    buildsystem: meson
+    make-args: [modules/printbackends/libprintbackend-cups.so]
+    no-make-install: true
+    post-install:
+      - install -Dm 755 modules/printbackends/libprintbackend-cups.so -t /app/lib/gtkmodules/3.0.0/printbackends/
+    sources:
+      - type: git
+        url: https://gitlab.gnome.org/GNOME/gtk.git
+        tag: 3.24.23
+
+  - name: gtk-settings
+    buildsystem: simple
+    build-commands:
+      - 'install -Dm 644 gtk-settings.ini /app/etc/xdg/gtk-3.0/settings.ini'
+    sources:
+      - type: file
+        path: gtk-settings.ini
+
+  - name: zypak
+    sources:
+      - type: git
+        url: https://github.com/refi64/zypak
+        branch: master
+        commit: 61be988e5de65aaec4d562db3d9c9c74357f6b01
+        # tag: v2020.10beta.3
+      # - type: dir
+      #   path: ../../code/zypak
+
+  - name: ungoogled-chromium
+    buildsystem: simple
+    build-commands:
+      - 'install -Dm 755 chromium.sh /app/bin/chromium'
+      - 'install -Dm 755 update-desktop-database /app/bin/update-desktop-database'
+      - 'install -Dm 755 xdg-icon-resource /app/bin/xdg-icon-resource'
+      - 'install -Dm 755 xdg-desktop-menu /app/bin/xdg-desktop-menu'
+      - 'install -Dm 644 -t /app/share/flatpak-chromium mimic_warning.html'
+      - 'install -Dm 644 -t /app/share/applications com.github.Eloston.ungoogled-chromium.desktop'
+      - 'install -Dm 644 -t /app/share/metainfo com.github.Eloston.ungoogled-chromium.metainfo.xml'
+      - 'install -Dm 644 chromium.svg /app/share/icons/hicolor/scalable/apps/com.github.Eloston.ungoogled-chromium.svg'
+
+      # Step1
+      - 'mkdir -p build/download_cache'
+      - 'mv chromium-85.0.4183.102.tar.xz build/download_cache/'
+      - 'mv chromium-85.0.4183.102.tar.xz.hashes build/download_cache/'
+      - './utils/downloads.py retrieve -c build/download_cache -i downloads.ini'
+      - './utils/downloads.py unpack -c build/download_cache -i downloads.ini -- build/src'
+
+      # Step2
+      - './utils/prune_binaries.py build/src pruning.list'
+
+      # Step3
+      - './utils/patches.py apply build/src patches'
+
+      # Step4
+      - './utils/domain_substitution.py apply -r domain_regex.list -f domain_substitution.list -c build/domsubcache.tar.gz build/src'
+
+      # Step5
+      - 'mkdir -p build/src/out/Default'
+      - 'cd build/src && ./tools/gn/bootstrap/bootstrap.py --skip-generate-buildfiles -j4 -o out/Default/'
+
+      # Step6
+      - 'mkdir -p build/src/out/Default'
+      - 'cp flags.gn build/src/out/Default/args.gn'
+      - 'echo use_sysroot=false >> build/src/out/Default/args.gn'
+      - 'cd build/src && ./out/Default/gn gen out/Default --fail-on-unused-args'
+      - 'cd build/src && ninja -C out/Default chrome chromedriver chrome_sandbox'
```

Comment:
Is it possible to use their [building process](https://github.com/Eloston/ungoogled-chromium/blob/master/docs/building.md#integrating-ungoogled-chromium-into-your-chromium-build-process) as a reference? As far as I understand, the yaml is downloading the source code of Chromium and then applying the ungoogled-chromium patches, which is what this section of the build process is about.

Reply:
^ he has a point. I can't see if patches are being applied. Hmm.

Reply:
Not possible because flatpak-builder doesn't allow internet connectivity which is what their build process requires

Reply:
Damn, that sucks

Reply:
but I am using the build process. Only difference is I downloaded the files ahead of time and put them in the right location so they count as "cached"

Reply:
It's the same exact commands except that sysroot is not a possibility which isn't a bit deal at all

Reply:
I'm not really good with Chromium, but does Chromium support ninja? Maybe running cmake only might work.

Reply:
@TheEvilSkeleton wouldn't that make the build even slower?

Reply:
Sorry for the late reply, but yes, that makes it slower. Not every program supports ninja, which is why you could try using cmake... if you're about to give up. @refi64 could confirm if Chromium supports ninja or not.

Reply:
err Chromium doesn't use CMake at all, it uses a custom build system that exclusively outputs Ninja build files, as being used right now.

Reply:
Compiling works now :)